src/test_l2_hdmi_cec_driver.c

```diff
@@ -1,77 +1,332 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+ 
 /**
-*  If not stated otherwise in this file or this component's LICENSE
-*  file the following copyright and licenses apply:
+* @file test_l2_hdmi_cec_driver.c
+* @page hdmi_cec_driver Level 2 Tests
+*
+* ## Module's Role
+* This module includes Level 2 functional tests (success and failure scenarios).
+* This is to ensure that the hdmi_cec_driver APIs meet the requirements across all vendors.
+*
+* **Pre-Conditions:**  None@n
+* **Dependencies:** None@n
 *
-*  Copyright 2023 RDK Management
+* Ref to API Definition specification documentation : [halSpec.md](../../../docs/halSpec.md)
+*/
+
+#include <ut.h>
+#include <ut_log.h>
+#include "hdmi_cec_driver.h"
+
+#define CEC_GET_CEC_VERSION (0x9F)
+
+/**
+* @brief This test aims to verify the functionality of adding and getting logical address in HDMI CEC driver
 *
-*  Licensed under the Apache License, Version 2.0 (the License);
-*  you may not use this file except in compliance with the License.
-*  You may obtain a copy of the License at
+* In this test, we open the HDMI CEC HAL, add a logical address, get the logical address and then close the HDMI CEC HAL. The test ensures that the add and get operations are successful and the logical address retrieved is the same as the one added.
 *
-*  http://www.apache.org/licenses/LICENSE-2.0
+* **Test Group ID:** 02@n
+* **Test Case ID:** 001@n
 *
-*  Unless required by applicable law or agreed to in writing, software
-*  distributed under the License is distributed on an AS IS BASIS,
-*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-*  See the License for the specific language governing permissions and
-*  limitations under the License.
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_hdmi_cec_driver_test_specification.md](l2_hdmi_cec_driver_test_specification.md)
 */
+void test_l2_hdmi_cec_driver_AddAndGetLogicalAddress(void)
+{
+    UT_LOG("Entering test_l2_hdmi_cec_driver_AddAndGetLogicalAddress...");
+
+    int handle;
+    int logicalAddress = 0x0;
+    int getLogicalAddress;
+    HDMI_CEC_STATUS status;
+
+    // Open HDMI CEC HAL
+    status = HdmiCecOpen(&handle);
+    UT_LOG(" HdmiCecOpen returned %d", status);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    // Add Logical Address
+    status = HdmiCecAddLogicalAddress(handle, logicalAddress);
+    UT_LOG("Invoking HdmiCecAddLogicalAddress with handle and logical address. Handle: %d, Logical Address: %d", handle, logicalAddress);
+    if (status != HDMI_CEC_IO_SUCCESS) 
+    {
+        UT_LOG("HdmiCecAddLogicalAddress failed with status: %d", status);
+        UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
```

Comment:
When CEC Add logical address fails, HDMICECclose is never called as the test exits from here.  Similar issues all through the file.  please check.

Reply:
done